string-match-guards for crafted-init-config

UristMcLisper opened the pull request:
The functionality for ignoring the auto-insert on the custom file may run into an issue 
when the `custom-file` and/or `file-buffer-name` values are `nil`.

This essentially breaks the `auto-insert` function for non-file buffers  (and all buffers 
if `custom-file` is `nil`).

jvdydev commented:
Is the commit author just an error in a git config (not the same username)?

Additionally, it would be nice to add some comment on *why* this is changed. 
It's much easier for us to figure out what you want changed (especially if there is no issue attached).
Doesn't have to be long, something like:
```markdown
The functionality for ignoring the auto-insert on the custom file may run into an issue 
when the `custom-file` and/or `file-buffer-name` values are `nil`.

This essentially breaks the `auto-insert` function for non-file buffers  (and all buffers 
if `custom-file` is `nil`).
```

UristMcLisper commented:
> Is the commit author just an error in a git config (not the same username)?
That's intentional. So that guix people can assign copyright based on git name ;)

UristMcLisper commented:
1. Follow getting started from README.org
2. Emacs starts with failure and any find-file displays buffer only on second try. 
3. I added check for custom-file
4. Same thing happened later when I tried to open .epub file usinv nov.el package. 
5. I added second check

Nothing to document I think

jeffbowman commented:
Tried those steps, everything worked perfectly for me.  I'll merge this anyway, it doesn't do any harm, and if it helps someone else, then we still win. 

Thanks again!
